internal/api/event.go

```diff
@@ -46,18 +46,18 @@ func FormatEventKubernetesUserActor(u authnv1.UserInfo) string {
 	return kargoapi.EventActorKubernetesUserPrefix + u.Username
 }
 
-func NewFreightApprovedEventAnnotations(actor string, f *kargoapi.Freight, stageName string) map[string]string {
-	annotations := map[string]string{
-		kargoapi.AnnotationKeyEventProject:           f.Namespace,
-		kargoapi.AnnotationKeyEventFreightCreateTime: f.CreationTimestamp.Format(time.RFC3339),
-		kargoapi.AnnotationKeyEventFreightAlias:      f.Alias,
-		kargoapi.AnnotationKeyEventFreightName:       f.Name,
-		kargoapi.AnnotationKeyEventStageName:         stageName,
+func NewFreightApprovedEvent(actor string, f *kargoapi.Freight, stageName string) event.PromotionEvent {
```

Comment:
I could also split this out as its own event if we prefer, rather than using a single event type to rule them all. I figured I'd keep it mostly the same as it currently is though

Reply:
Answering my own question here: After looking at it more, I think we do want two different events. There will be duplicate fields between them, but they will work better as separate events